src/snippet.rs

```diff
@@ -149,32 +220,40 @@ impl<'a> Snippet<'a> {
         self
     }
 
-    pub fn annotation(mut self, annotation: Annotation<'a>) -> Self {
-        self.annotations.push(annotation);
+    pub fn fold(mut self, fold: bool) -> Self {
+        self.fold = fold;
         self
     }
 
-    pub fn annotations(mut self, annotation: impl IntoIterator<Item = Annotation<'a>>) -> Self {
-        self.annotations.extend(annotation);
+    pub fn col_separator(mut self, col_separator: bool) -> Self {
+        self.col_separator = Some(col_separator);
         self
     }
+}
 
-    /// Hide lines without [`Annotation`]s
-    pub fn fold(mut self, fold: bool) -> Self {
-        self.fold = fold;
+impl<'a> Snippet<'a, Annotation<'a>> {
+    pub fn annotation(mut self, annotation: Annotation<'a>) -> Snippet<'a, Annotation<'a>> {
+        self.markers.push(annotation);
+        self
+    }
+
+    pub fn annotations(mut self, annotation: impl IntoIterator<Item = Annotation<'a>>) -> Self {
+        self.markers.extend(annotation);
         self
     }
 }
 
-/// An annotation for a [`Snippet`].
-///
-/// See [`Level::span`] to create a [`Annotation`]
+impl<'a> From<Snippet<'a, Annotation<'a>>> for Element<'a> {
+    fn from(value: Snippet<'a, Annotation<'a>>) -> Self {
+        Element::Cause(value)
+    }
+}
```

Comment:
I'd put all of these `From`s next to `Element` as this is about defining how to interact with an `Element` and not each of the other types